source/common/http/utility.cc

```diff
@@ -288,5 +288,15 @@ const std::string& Utility::getProtocolString(const Protocol protocol) {
   NOT_REACHED;
 }
 
+void Utility::appendToHeader(HeaderString& header, const std::string& data) {
+  if (data.empty()) {
+    return;
+  }
+  if (!header.empty()) {
+    header.append(",", 1);
+  }
+  header.append(data.c_str(), data.size());
```

Comment:
The perf-optimizer in me so wants to avoid the double append but having a utility for this is one step in the right direction and we can get around to optimizing it later :-P
@jmarantz can add it to his ever-expanding list if it ever shows up on profiles ;-)

Reply:
I just checked the `HeaderString` implementation, and it uses a power-of-two buffer allocator, so the double-append might not be that bad in practice.

Reply:
yeah, if we append something short to something long we're fine. So
reallylongthing -> reallylongthing,addition is fine but I believe
short -> short,reallylongaddition will result in a double move for the comma and the long addition.

It's no worse than things were before, and now we have a utility we can optimize more easily whenever we get around to caring so I think we're good for this PR :-)